src/common/models/app-settings/app-settings.ts

```diff
@@ -15,247 +15,167 @@
-    } = parameters;
-
-    for (const dataCube of dataCubes) {
-      if (dataCube.clusterName === "native") continue;
-      if (!NamedArray.findByName(clusters, dataCube.clusterName)) {
-        throw new Error(`data cube ${dataCube.name} refers to an unknown cluster ${dataCube.clusterName}`);
-      }
-    }
-
-    this.clientTimeout = clientTimeout || DEFAULT_CLIENT_TIMEOUT;
-    this.version = version || 0;
-    this.clusters = clusters;
-    this.customization = customization;
-    this.dataCubes = dataCubes;
-  }
-
-  public valueOf(): AppSettingsValue {
-    return {
-      version: this.version,
-      clientTimeout: this.clientTimeout,
-      clusters: this.clusters,
-      customization: this.customization,
-      dataCubes: this.dataCubes
-    };
-  }
-
-  public toJS(): AppSettingsJS {
-    let js: AppSettingsJS = {};
-    if (this.version) js.version = this.version;
-    js.clientTimeout = this.clientTimeout;
-    js.clusters = this.clusters.map(cluster => cluster.toJS());
-    js.customization = this.customization.toJS();
-    js.dataCubes = this.dataCubes.map(dataCube => dataCube.toJS());
-    return js;
-  }
-
-  public toJSON(): AppSettingsJS {
-    return this.toJS();
-  }
-
-  public toString(): string {
-    return `[AppSettings v${this.version} dataCubes=${this.dataCubes.length}]`;
-  }
-
-  public equals(other: AppSettings): boolean {
-    return AppSettings.isAppSettings(other) &&
-      this.version === other.version &&
-      this.clientTimeout === other.clientTimeout &&
-      immutableArraysEqual(this.clusters, other.clusters) &&
-      immutableEqual(this.customization, other.customization) &&
-      immutableArraysEqual(this.dataCubes, other.dataCubes);
-  }
-
-  public toClientSettings(): AppSettings {
-    let value = this.valueOf();
-
-    value.clusters = value.clusters.map(c => c.toClientCluster());
-
-    value.dataCubes = value.dataCubes
-      .filter(ds => ds.isQueryable())
-      .map(ds => ds.toClientDataCube());
-
-    return new AppSettings(value);
-  }
-
-  public getVersion(): number {
-    return this.version;
-  }
-
-  public getDataCubesForCluster(clusterName: string): DataCube[] {
-    return this.dataCubes.filter(dataCube => dataCube.clusterName === clusterName);
-  }
-
-  public getDataCube(dataCubeName: string): DataCube {
-    return NamedArray.findByName(this.dataCubes, dataCubeName);
-  }
-
-  public addOrUpdateDataCube(dataCube: DataCube): AppSettings {
-    let value = this.valueOf();
-    value.dataCubes = NamedArray.overrideByName(value.dataCubes, dataCube);
-    return new AppSettings(value);
-  }
-
-  public deleteDataCube(dataCube: DataCube): AppSettings {
-    let value = this.valueOf();
-    const index = value.dataCubes.indexOf(dataCube);
-
-    if (index === -1) {
-      throw new Error(`Unknown dataCube : ${dataCube.toString()}`);
-    }
-
-    let newDataCubes = value.dataCubes.concat();
-    newDataCubes.splice(index, 1);
+export interface SerializedClientAppSettings {
```

Comment:
I'm not sure that app-settings is the best place for client app settings interfaces

Reply:
`AppSettings` before was one thing, now I extracted few interfaces but they are mostly the same thing. Just differ in some modifiers on fields. I think they could be in one file, especially that there are function that translate between two types.

Reply:
I understand that app-settings cover server and client settings but this is a lot of code at the end.